diff --git a/node_modules/react-native-block-store/android/src/main/java/com/blockstore/BlockStoreModule.kt b/node_modules/react-native-block-store/android/src/main/java/com/blockstore/BlockStoreModule.kt
index 80abbb0..ccdb00b 100644
--- a/node_modules/react-native-block-store/android/src/main/java/com/blockstore/BlockStoreModule.kt
+++ b/node_modules/react-native-block-store/android/src/main/java/com/blockstore/BlockStoreModule.kt
@@ -40,7 +40,16 @@ class BlockStoreModule internal constructor(context: ReactApplicationContext) :
 
     blockStoreClient.retrieveBytes(retrieveRequest)
       .addOnSuccessListener {
-        promise.resolve(if (it.blockstoreDataMap.containsKey(key)) it.blockstoreDataMap[key] else null)
+        val dataMap = it.blockstoreDataMap
+        if (dataMap.containsKey(key)) {
+          val data = dataMap[key]
+          if (data != null) {
+            promise.resolve(String(data.bytes, Charsets.UTF_8))
+          } else {
+            promise.resolve(null)
+          }
+        } else {
+          promise.resolve(null)
+        }
       }
       .addOnFailureListener {
         promise.resolve(null)
diff --git a/node_modules/react-native-block-store/android/src/oldarch/BlockStoreSpec.kt b/node_modules/react-native-block-store/android/src/oldarch/BlockStoreSpec.kt
index 52d0c1d..da3d9a0 100644
--- a/node_modules/react-native-block-store/android/src/oldarch/BlockStoreSpec.kt
+++ b/node_modules/react-native-block-store/android/src/oldarch/BlockStoreSpec.kt
@@ -7,9 +7,9 @@ import com.facebook.react.bridge.Promise
 abstract class BlockStoreSpec internal constructor(context: ReactApplicationContext) :
   ReactContextBaseJavaModule(context) {
 
-  abstract fun save(key: String, value: String, shouldBackupToCloud: boolean, promise: Promise);
+  abstract fun save(key: String, value: String, shouldBackupToCloud: Boolean, promise: Promise)
 
-  abstract fun retrieve(key: String, promise: Promise);
+  abstract fun retrieve(key: String, promise: Promise)
 
-  abstract fun remove(key: String, promise: Promise);
+  abstract fun remove(key: String, promise: Promise)
 }
